### ISSUE TITLE
GDrive backups

### ISSUE THREAD

**Issue by Osiris-Team** [Osiris-Team/AutoPlug-Client#12]

**Comment by fulcanelly** [comment]
Backups of what exactly, world or plugins ?

**Comment by Osiris-Team** [comment]
@fulcanelly Currently autoplug creates one backup zip for the complete server. The user can exclude/include specific files/folders. It would be great to be able to upload those backups to google drive and also handle the deletion of older ones on google drive.

**Comment by jivanpal** [comment]
I cam here wondering if there was an existing feature request for custom backup scripts. Seeing this, I am now wondering whether that would be in scope here as a new backup mechanism, e.g. we could implement a `backup-mechanism` setting whose values could be `local-zip` (default, current behaviour), `local`, `google-drive`, other mechanisms desired down the line, or `custom-script` (which I'm interested in, as I would like to just take [Restic](https://restic.net/) snapshots of the directory whilst the server is down, without having to write an even more complex script to interact with AutoPlug in a tmux session in order to bring the server down, do the backup, and then bring the server back up).

If that sounds desirable, perhaps it would be better to implement `custom-script` first, and then just have other mechanisms such as `google-drive` be aliases for predefined scripts?

**Comment by Osiris-Team** [comment]
@jivanpal Yeah that sounds good, you can create a new issue for `custom-script`.
If you know how to code in Java or know someone feel free to create a pull request.

**Comment by jivanpal** [comment]
Wow, quick response!

> Note that you would probably need to exclude the AutoPlug-Client.jar in your alternative backup software, since autoplug is still running at that time.

Surely AutoPlug-Client.jar is not changing during backup? The self-updater cannot run at the same time as the backup task, can it? From my logs, I was under the impression that the tasks run sequentially, not concurrently.

**Comment by Osiris-Team** [comment]
Yeah, AutoPlug is **not** changing. When running the tasks, all other tasks wait for the backup task to finish and then run concurrently.

**Comment by JustABiologist** [comment]
Implemented this in PR #279: https://github.com/Osiris-Team/AutoPlug-Client/pull/279\n\nPer the bounty terms, if this is accepted/merged, PayPal payout destination: stab.me.papi@gmail.com

**Comment by Zoriot** [comment]
@Osiris-Team can you explain what implementation is eligible for a bounty and the size of the bounry? 

I saw in https://github.com/Osiris-Team/AutoPlug-Client/pull/279#issuecomment-4440129925 you pointed out it may be partly eligible if these things are provided. 
Can you clearify that a bit more?

**Comment by Osiris-Team** [comment]
already implemented
